builtin/providers/aws/resource_aws_key_pair.go

```diff
@@ -65,6 +65,11 @@ func resourceAwsKeyPairRead(d *schema.ResourceData, meta interface{}) error {
 	}
 	resp, err := conn.DescribeKeyPairs(req)
 	if err != nil {
+		awsErr, ok := err.(aws.APIError)
+		if ok && awsErr.Code == "InvalidKeyPair.NotFound" {
```

Comment:
What do you think about establishing a precedent to log a warning when we do `d.SetId("")` whenever we're in this `NotFound` situation? I imagine we don't so that in many places yet, or at all, but I feel like we should.